guides/get-started-microservices-on-kubernetes/simple-microservice-database-part1.adoc

```diff
@@ -0,0 +1,413 @@
+
+== Maven Project
+
+You will extend the sample application you created in link:simple-microservice-part1[{simple-microservice-part1}] by adding the link:https://github.com/wildfly-extras/wildfly-datasources-galleon-pack[wildfly-datasources-galleon-pack];
+
+The link:https://github.com/wildfly-extras/wildfly-datasources-galleon-pack[wildfly-datasources-galleon-pack] will:
+
+* install the JDBC Driver for the PostgreSQL Database into WildFly
+* add a database connection to PostgreSQL in WildFly
+
+=== pom.xml
+
+==== dependencies
+
+Add the following dependencies to the `pom-xml` file `dependencies` section:
+
+[source,xml,subs="normal"]
+----
+        <dependency>
+            <groupId>jakarta.persistence</groupId>
+            <artifactId>jakarta.persistence-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.xml.bind</groupId>
+            <artifactId>jakarta.xml.bind-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.transaction</groupId>
+            <artifactId>jakarta.transaction-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+----
+
+==== wildfly-maven-plugin
+
+Add the `wildfly-datasources-galleon-pack` feature-pack and the `postgresql-datasource` layer to the `wildfly-maven-plugin` configuration;
+
+You should end up with the link:https://github.com/wildfly/wildfly-maven-plugin/[wildfly-maven-plugin] configured like in the following:
+
+[source,xml,subs="normal"]
+----
+    <plugin>
+        <groupId>org.wildfly.plugins</groupId>
+        <artifactId>wildfly-maven-plugin</artifactId>
+        <version>{version-wildfly-maven-plugin}</version>
+        <configuration>
+            <feature-packs>
+                <feature-pack>
+                    <location>org.wildfly:wildfly-galleon-pack:{version-wildfly-galleon-pack}</location>
+                </feature-pack>
+                <feature-pack>
+                    <location>org.wildfly.cloud:wildfly-cloud-galleon-pack:{version-wildfly-cloud-galleon-pack}</location>
+                </feature-pack>
+                <feature-pack>
+                    <location>org.wildfly:wildfly-datasources-galleon-pack:{version-wildfly-datasources-galleon-pack}</location>
+                </feature-pack>
+            </feature-packs>
+            <layers>
+                <layer>cloud-server</layer>
+                <layer>postgresql-datasource</layer>
+            </layers>
+        </configuration>
+        <executions>
+            <execution>
+                <goals>
+                    <goal>package</goal>
+                </goals>
+            </execution>
+        </executions>
+    </plugin>
+----
+
+=== Java Classes
+
+Add the following classes to the project:
+
+.org.wildfly.examples.model.TestTable   :
+[source,java]
+----
+package org.wildfly.examples.model;
```

Comment:
Instead of providing packages for model,service,repository ..., it is better to use a domain driven structure. If you resource is named "TestTable", then add a test table package and there and the service, repository, model for such a resource. This is just a simple guide, but this is a better practice for project structures.

Reply:
Thank you @yersan I did not add repository but refactored the two classes as you suggested

Reply:
Sorry if it was not clear. You should not create service, repository, model subpakages inside, just place the related business domain objects in the same package, e.g test_table, but do not create subpackages as test_table.service, test_table.repository, test_table.model ... otherwise we will be in the same case as before.

Reply:
@yersan tnks, moved both `TestTable` and `TestTableResource` to package `org.wildfly.examples.test_table`